pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -304,11 +304,37 @@ func SetDefaults_Shoot(obj *Shoot) {
 			continue
 		}
 
-		if worker.CRI != nil {
+		if isDockerRuntime(worker.CRI) {
 			continue
 		}
 
-		obj.Spec.Provider.Workers[i].CRI = &CRI{Name: CRINameContainerD}
+		if worker.CRI == nil {
+			obj.Spec.Provider.Workers[i].CRI = &CRI{Name: CRINameContainerD}
+		}
+
+		// When CRI runtime is used and there is no explicit kubelet configuration, the ContainerLogMaxSize in Workers kubelet is set to 10Mi.
+		// To align both container runtime configurations, the default log max size in containerd case is also set to 100Mi.
+		r := resource.MustParse(string(DefaultContainerLogMaxSize))
+
+		if worker.Kubernetes == nil {
+			obj.Spec.Provider.Workers[i].Kubernetes = &WorkerKubernetes{
+				Kubelet: &KubeletConfig{
+					ContainerLogMaxSize: &r,
+				},
+			}
+			continue
+		}
+
+		if worker.Kubernetes.Kubelet == nil {
+			obj.Spec.Provider.Workers[i].Kubernetes.Kubelet = &KubeletConfig{
+				ContainerLogMaxSize: &r,
+			}
+			continue
+		}
+
+		if worker.Kubernetes.Kubelet.ContainerLogMaxSize == nil {
+			obj.Spec.Provider.Workers[i].Kubernetes.Kubelet.ContainerLogMaxSize = &r
+		}
```

Comment:
Can this be written as follows

```suggestion
		if worker.Kubernetes == nil {
			obj.Spec.Provider.Workers[i].Kubernetes = &WorkerKubernetes{}
		}

		if worker.Kubernetes.Kubelet == nil {
			obj.Spec.Provider.Workers[i].Kubernetes.Kubelet = &KubeletConfig{}
		}

		if worker.Kubernetes.Kubelet.ContainerLogMaxSize == nil {
			defaultContainerLogMaxSize := resource.MustParse(string(DefaultContainerLogMaxSize))
			obj.Spec.Provider.Workers[i].Kubernetes.Kubelet.ContainerLogMaxSize = &defaultContainerLogMaxSize
		}
```

?

Reply:
Did you run the suggestion?

Reply:
The approach here is crystal clear what it defaults, why do you want to change it?

Reply:
> Did you run the suggestion?

Yes, it should do the same thing - default `.kubernetes.kubelet.containerLogMaxSize` to `100Mi` for a worker pool when the corresponding field is not set.

> The approach here is crystal clear what it defaults, why do you want to change it?

I am not saying that the approach is unclear or wrong. I was just suggesting the another one because I feel that it is used more often across the code base:
https://github.com/gardener/gardener/blob/5bf28f8ff7ecf3e2ffe21224f0cb6ee30daf9997/pkg/apis/core/v1alpha1/defaults.go#L314-L325

https://github.com/gardener/gardener/blob/5bf28f8ff7ecf3e2ffe21224f0cb6ee30daf9997/pkg/apis/core/v1alpha1/defaults.go#L203-L209

Reply:
Please respect peoples' work and make suggestions when this leads to real improvement. Further more if you look at the original code you will see similar approach there.

Reply:
TBH, I agree with @ialidzhikov suggestion here. This is what we do elsewhere as well and allows to extend such constructs in the future easier/with less confusion (think about a new field in the `KubeletConfiguration` that we'd like to default here). With the current approach, we have to assign this at three places which can quickly blow up.

I don't think this is about not respecting your work but rather about consistency across the code base and best practices derived from experiences in the past years.
It might look minor and it certainly is, but there was a reason behind this suggestion.